### DISCUSSION TITLE
Something went wrong when I read the result from yalmip

### DISCUSSION THREAD

**Discussion by zjuyoung** [yalmip/YALMIP#1055] [Problems using YALMIP] [answered]
Thank you for your attention.
I built a Simulink project to solve an economic MPC problem. The controller was designed first when time=0 as the demo shows.  As shown in the DMC_SLW.m file,  I defined three variables _POWER_PRI/TEMP_PRI/EX_ELC_PRICE_ as inputs, however, when I read them from the result, I found the values were totally different from the values I defined. And I want to know how to correct it, thank you!
One can run the A_mpc_design.m file first and then run the dmctest.slx file, I have set a breakpoint at the DMC_SLW.m，you can check the output result of the controller, _p_, _t_, and _price_  which are different from the defined values _power_pri_, _temp_pri_, and _price_pri_. Thank you for your help.
   @johanlofberg [files.zip](https://github.com/yalmip/YALMIP/files/7750856/files.zip)

**Comment by johanlofberg** [comment]
That is a way too complex setup for me to look into

However, the question is not clear. You say you define variables (I presume you mean decision variables, i.e. sdpvars). Then you say they have values different from the ones you have defined. I presume you mean when you solve the optimization problem, the values that these decision variables take, do not satisfy the constraints you have imposed on them. If that is the case, I would guess the problem has not really been solved (have you looked at the diagnostic code and the solver display), or they are just marginally infeasible and you have to high expectations on the numerical precision of a solver.

And as I always say, never use optimizer before you know everything works with optimize, and once switching to optimizer, you are promising optimizer that the model you are defining is in a structural form that the solver you have specified actually can solve the problem in that form once the parameters are fixed, and you always use optimizer with verbose set to 2 when starting using it.

**Comment by johanlofberg** [comment, marked as answer]
and your code appears to use some stuff both as parameters and outputs in optimizer. that is not supported. parameters cannot be returned in the output list (well they can, but expected behavior is undefined)

**Reply by zjuyoung** [reply]
Thank you for your precious advice dear professor, I have corrected the error, just as you said, the parameter can not be returned in the output list, thank you very much!
